models/genmodels.go

```diff
@@ -324,16 +266,50 @@ var typeTemplate = `
 	// {{.Output}} defines the {{.StatusCode}} status code response for {{.OpName}}.
 	type {{.Output}} {{.Type}}
 
-	{{if .ErrorType}}
-	// Error returns "Status Code: X". We implemented in to satisfy the error
-	// interface. For a more descriptive error message see the output type.
-	func (o {{.Output}}) Error() string {
-		return "Status Code: {{.StatusCode}}"
-	}
-	{{else}}
 	// {{.OpName}}StatusCode returns the status code for the operation.
 	func (o {{.Output}}) {{.OpName}}StatusCode() int {
 		return {{.StatusCode}}
 	}
-	{{end}}
 `
+
+// generateErrorMethods finds all responses all error responses and generates an error
+// method for them.
+func generateErrorMethods(s *spec.Swagger) (string, error) {
+	errorTypes := stringset.New()
+
+	for _, pathKey := range swagger.SortedPathItemKeys(s.Paths.Paths) {
```

Comment:
you can pull out this code into a helper function if it's useful for the JS client